components/test-file-results.html

```diff
@@ -0,0 +1,168 @@
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+<link rel="import" href="/bower_components/polymer/polymer-element.html">
+<link rel="import" href="/bower_components/polymer/lib/elements/dom-if.html">
+<link rel="import" href="/bower_components/polymer/lib/elements/dom-repeat.html">
+<link rel="import" href="/components/test-run.html">
+
+<dom-module id="test-file-results">
+  <template>
+    <style>
+      :host {
+        display: block;
+        font-size: 16px;
+      }
+      h1 {
+        font-size: 1.5em;
+      }
+
+      table {
+        width: 100%;
+        border-collapse: collapse;
+      }
+      td.sub-test-name {
+        font-family: monospace;
+        font-size: 0.9em;
+      }
+      td.result {
+        background-color: #eee;
+      }
+      td.result.OK, td.result.PASS {
+        background-color: rgb(90, 242, 113);
+      }
+      td.result.FAIL {
+        background-color: rgb(242, 90, 90);
+      }
+    </style>
+
+    <table>
+      <thead>
+        <tr>
+          <th>Subtest</th>
+          <template is="dom-repeat" items="{{testRuns}}" as="testRun">
+            <th><test-run test-run="[[testRun]]"></test-run></th>
+          </template>
+        </tr>
+      </thead>
+      <tbody>
+        <template is="dom-repeat" items="{{subTestNames}}" as="subtestName">
+          <tr>
+            <td class="sub-test-name">{{ subtestName }}</td>
+
+            <template is="dom-repeat" items="{{testRuns}}" as="testRun">
+              <td class$="result [[ _subTestResultForTestRun(testRun, subtestName) ]]">
+                {{ _subTestResultForTestRun(testRun, subtestName) }}
+              </td>
+            </template>
+          </tr>
+        </template>
+
+      </tbody>
+    </table>
+  </template>
+
+  <script>
+    class TestFileResults extends window.Polymer.Element {
+      static get is () { return 'test-file-results' }
+
+      static get properties () {
+        return {
+          testRuns: {
+            type: Array
+          },
+          testFile: {
+            type: String,
+            observer: '_testFileChanged'
+          },
+          subTestNames: {
+            type: Array,
+            value: []
+          }
+        }
+      }
+
+      connectedCallback () {
+        super.connectedCallback()
+        console.assert(this.testFile)
+        console.assert(this.testFile[0] === '/')
+        console.assert(this.testRuns)
+        console.assert(this.testRuns.length > 0)
+      }
+
+      async fetchTestFile () {
```

Comment:
Is one test file one JSON file stored in cloud storage which contains all results for all runs for a given test? And when viewing results for one test (including maybe multiple test runs on multiple browsers), there's one fetch?

What are you thoughts about "API documentation" or function summary comments for functions like this? Helpful or no? Some projects use structured comments like JSDoc, but others just have occasional one-or-two-line comments above the function.